drivers/ma_reports/app/models/ma_reports/monthly_performance/report.rb

```diff
@@ -334,14 +338,118 @@ def demographic_breakdowns
       end
     end
 
+    def race_thnicity_client_breakdowns
+      @race_thnicity_client_breakdowns ||= calculate_race_ethnicities
+    end
+
+    def calculate_race_ethnicities
+      race_ethnicities_breakdown = race_ethnicity_combinations.values.map { |m| [m, Set[]] }.to_h
+      scope = enrollments
+      race_data = pluck_to_hash((['id'] + race_columns.excluding('race_none')).map { |k, v| [k, v] }.to_h, scope)
+      race_ethnicity_combinations.each do |key, value|
+        lookups[key]&.call(race_data, race_ethnicities_breakdown[value])
+      end
+      race_ethnicities_breakdown
+    end
+
+    def race_col_lookup
```

Comment:
Let's centralize this in HudUtility2024 and remove/replace from `drivers/system_pathways/app/models/system_pathways/report.rb` as well.